File: src/main/kotlin/tech/diggle/apps/accomodation/elections/ElectionServiceImpl.java
package tech.diggle.apps.accomodation.elections;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Service;

import java.util.Date;
import java.util.List;


@Service
public class ElectionServiceImpl  implements ElectionService{

    @Autowired
    @Qualifier("electionRepository")
    ElectionRepository electionRepository;



    @Override
    public List<Election> listAll() {
        return  electionRepository.findAll();
    }

    @Override
    public Election createElection(Election election) {

        election.setElectionDate(election.getElectionDate());
        election.setElectionType(election.getElectionType());
        electionRepository.save(election);
        return election;
    }

    @Override
    public boolean deleteElection(Election election) {
        return false;
    }

    @Override
    public Date findElectionByElectionDate(Date election) {
        return electionRepository.findElectionByElectionDate(election);
    }

    @Override
    public Election findElectionByElectionType(String election) {
        return electionRepository.findElectionByElectionType(election);
    }
}

File: src/main/kotlin/tech/diggle/apps/accomodation/election/ElectionServiceImpl.kt
package tech.diggle.apps.accomodation.election

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ElectionServiceImpl(@Autowired val repository: ElectionRepository) : ElectionService {
    override fun getAll(): List<Election> = repository.findAll()

    override fun get(id: Long): Election = repository.getOne(id)

    override fun add(election: Election): Election {
        return this.repository.save(election)
    }

    override fun update(election: Election): Election {
        return this.repository.save(election)
    }

    override fun find(election: Election): Election? {
        return this.repository.findByDateAndType(election.date!!, election.type!!)
    }

}